tools/c7n_tencentcloud/c7n_tencentcloud/resources/cam.py

```diff
@@ -0,0 +1,473 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+from concurrent.futures import as_completed
+import json
+import pytz
+from c7n.exceptions import PolicyValidationError
+from c7n.utils import chunks, type_schema
+from c7n_tencentcloud.provider import resources
+from c7n_tencentcloud.query import ResourceTypeInfo, QueryResourceManager, DescribeSource
+from c7n_tencentcloud.utils import isoformat_datetime_str, PageMethod, convert_date_str
+from c7n.filters import ValueFilter, Filter
+from c7n_tencentcloud.actions.core import TencentCloudBaseAction
+
+
+class DescribeCAM(DescribeSource):
+    """DescribeCAM"""
+    def get_resource_qcs(self, resources):
+        qcs_list = []
+
+        # for CAM, there is no region in qcs
+        for r in resources:
+            qcs = self.get_qcs(self.resource_type.service,
+                               "",
+                               self.resource_manager.config.account_id,
+                               self.resource_type.resource_prefix,
+                               r[self.resource_type.id])
+            qcs_list.append(qcs)
+        return qcs_list
+
+
+@resources.register("cam-user")
+class User(QueryResourceManager):
+    """
+    User - user management in Cloud Access Management (CAM)
+    https://www.tencentcloud.com/document/product/1021/37656
+    """
+    source_mapping = {"describe": DescribeCAM}
+
+    class resource_type(ResourceTypeInfo):
+        """resource_type"""
+        id = "Uin"
+        endpoint = "cam.tencentcloudapi.com"
+        service = "cam"
+        version = "2019-01-16"
+        enum_spec = ("ListUsers", "Response.Data[]", {})
+        taggable = True
+        resource_prefix = "uin"
+        batch_size = 10
+        datetime_fields_format = {
+            "CreateTime": ("%Y-%m-%d %H:%M:%S", pytz.timezone("Asia/Shanghai")),
+            "LastLoginTime": ("%Y-%m-%d %H:%M:%S", pytz.timezone("Asia/Shanghai")),
+            "access_keys.CreateTime": ("%Y-%m-%d %H:%M:%S", pytz.timezone("Asia/Shanghai")),
+            "access_keys.LastUsedDate": ("%Y-%m-%d", pytz.timezone("Asia/Shanghai"))
+        }
+
+    def __init__(self, ctx, data):
+        super().__init__(ctx, data)
+        self._query_client = None
+
+    @property
+    def client(self):
+        if self._query_client is None:
+            self._query_client = self.get_client()
+        return self._query_client
+
+    def augment(self, resources):
+        for item in resources:
+            item["password_enabled"] = not item["ConsoleLogin"] == 0
+            fm = self.resource_type.datetime_fields_format["CreateTime"]
+            item["CreateDate"] = isoformat_datetime_str(item["CreateTime"], fm[0], fm[1])
+        return resources
+
+    def set_user_login_mfa_active(self, resource):
+        """set_user_login_mfa_active"""
+        params = {"SubUin": resource[self.resource_type.id]}
+        resp = self.client.execute_query("DescribeSafeAuthFlagColl", params)
```

Comment:
this feels like it would be best left to a separate filter. so we don't have to query it if don't need it for the policy assertions, additionally its quite possible there are additional attributes the mfa device that want to be asserted, which a dedicated filter enables.

Reply:
1. moved these methods to filter
2. used a prefix key("c7n:") to store data